src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1445,6 +1466,32 @@ void MulticopterPositionControl::control_auto(float dt)
 	_limit_vel_xy = (!next_setpoint_valid || (_pos_sp_triplet.current.type == position_setpoint_s::SETPOINT_TYPE_LOITER))
 			&& (sqrtf(dist(0) * dist(0) + dist(1) * dist(1)) <= _target_threshold_xy.get());
 
+
+	if (_sonar_measurament.current_distance < _sonar_measurament.max_distance) {
+		if (!_vehicle_land_detected.landed && !in_auto_takeoff()) {
+			if (_obstacle_ahead_sp == false) {
+				stop_sp = _pos;
+				last_obstacle_dist = _sonar_measurament.current_distance;
+				_obstacle_ahead_sp = true;
+			}
+
+
+			if (_obstacle_ahead_sp == true && _sonar_measurament.current_distance < 0.6f) {
+				stop_sp(0) = prev_sp(0);
+				stop_sp(1) = prev_sp(1);
+			}
+
+			_curr_pos_sp(0) = stop_sp(0);
+			_curr_pos_sp(1) = stop_sp(1);
+		}
+
+	} else {
```

Comment:
If the range sensor maxes out (e.g. when there is no obstacle), you set the current position setpoint to the last `stop_sp` you computed. Doesn't this mean that the vehicle will be stopped indefinitely as soon as the range sensor has a single valid measurement (i.e. the if-statement starting at line 1472 is evaluated)?